server/lib/src/idm/oauth2.rs

```diff
@@ -874,6 +875,11 @@ impl Oauth2ResourceServersWriteTransaction<'_> {
                         }
                         false => None,
                     };
+
+                let consent_prompt_enabled = ent
+                    .get_ava_single_bool(Attribute::OAuth2ConsentPromptEnable)
+                    .unwrap_or(true);
```

Comment:
To answer your question re consent on public vs confidential clients: you can match on `type_` which tells you if the client is public. Then we allow this value to only go false on Basic clients.  

`let consent_prompt_enable == matches!(type_, OAuth2RSType::Public) || ent.get_ava .... `